File: crates/flowrt-codegen/tests/golden/bounded_operation_iox2_cpp/expected/app/stubs/cpp/controller.cpp
// FlowRT 管理参考模板（C++）。可删除重建；复制到用户 app/ 后再修改。

#include "flowrt_app/runtime_shell.hpp"

#include <memory>

namespace {

class Controller final : public flowrt_app::ControllerInterface {
public:
    flowrt::Status on_tick(
        flowrt_app::OperationClient_controller_plan& plan) override {
        return flowrt::ok();
    }
};

}  // namespace

namespace flowrt_user {

flowrt_app::App build_app() {
    return flowrt_app::App(std::make_unique<Controller>());
}

}  // namespace flowrt_user

File: crates/flowrt-codegen/tests/golden/zenoh_operation_rust/expected/app/stubs/rust/controller.rs
// FlowRT 管理参考模板（Rust）。可删除重建；复制到用户 app/ 后再修改。

#[derive(Default)]
pub struct Controller;

impl flowrt_app::components::Controller for Controller {
    fn on_tick(
        &mut self,
        _plan: &flowrt_app::components::OperationClient_controller_plan,
    ) -> flowrt::Status {
        flowrt::Status::Ok
    }
}

pub fn build_app() -> flowrt_app::runtime_shell::App {
    flowrt_app::runtime_shell::App::new(Box::new(Controller::default()))
}
